src/components/map/DrawingTools.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+
+import { Feature, Map } from 'ol';
+import { Geometry } from 'ol/geom';
+import VectorLayer from 'ol/layer/Vector';
+import VectorSource from 'ol/source/Vector';
+import { MdOutlineRectangle } from 'react-icons/md';
+
+import './styles.scss';
+
+type DrawingToolsProps = {
+  map: Map;
+  drawRectangle: (drawingSource: VectorSource<Feature<Geometry>>) => void;
+};
+
+export const DrawingTools = ({ map, drawRectangle }: DrawingToolsProps) => {
+  const [drawingSource, setDrawingSource] = useState<VectorSource<Feature<Geometry>> | undefined>(
```

Comment:
Is `useState` sufficient here? 

Perhaps I use context too much, but I could envisage a situation where there's a separate component that may need to activate the draw tools.

Reply:
I think you do, keep it simple until you need the complexity. If it had to be shared, then I'd still probably use `useState` but just in a re-usable hook instead, that can share state around, this is how zustrand and recoil work, which are alternatives to the likes of redux.